appinventor/components/src/com/google/appinventor/components/runtime/util/MediaUtil.java

```diff
@@ -316,29 +316,47 @@ public static BitmapDrawable getBitmapDrawable(Form form, String mediaPath) thro
 
     BitmapFactory.Options options;
     try {
-      options = getBitmapOptions(form, is1);
+      options = getBitmapOptions(form, is1, mediaPath);
     } finally {
       is1.close();
     }
 
     InputStream is2 = openMedia(form, mediaPath, mediaSource);
     try {
+      Log.d(LOG_TAG, "mediaPath = " + mediaPath);
       BitmapDrawable originalBitmapDrawable = new BitmapDrawable(decodeStream(is2, null, options));
-      // To be able to support different density devices, we might need to scale the bitmap
-      // The steps are:
+      // If options.inSampleSize == 1, then the image was not unreasonably large and may represent
+      // the actual size the user intended for the image. However we still have to scale it by
+      // the device density.
+      // However if we *did* sample the image to make it smaller, then that means that the image
+      // was not sized specifically for the application. In that case it makes no sense to
+      // scale it, so we don't.
+      // When we scale the image we do the following steps:
       //   1. set the density in the returned bitmap drawable.
       //   2. calculate scaled width and height
       //   3. create a scaled bitmap with the scaled measures
       //   4. create a new bitmap drawable with the scaled bitmap
       //   5. set the density in the scaled bitmap.
+
+      // Note: The log line below requires HoneycombMR1, but we may not keep it, so punting for now
+      Log.d(LOG_TAG, "originalBitmap.getByteCount() = " + originalBitmapDrawable.getBitmap().getByteCount());
+
       originalBitmapDrawable.setTargetDensity(form.getResources().getDisplayMetrics());
+      if (options.inSampleSize != 1) {
+        return originalBitmapDrawable;
+      }
       int scaledWidth = (int) (form.deviceDensity() * originalBitmapDrawable.getIntrinsicWidth());
       int scaledHeight = (int) (form.deviceDensity() * originalBitmapDrawable.getIntrinsicHeight());
+      Log.d(LOG_TAG, "form.deviceDensity() = " + form.deviceDensity());
+      Log.d(LOG_TAG, "originalBitmapDrawable.getIntrinsicWidth() = " + originalBitmapDrawable.getIntrinsicWidth());
+      Log.d(LOG_TAG, "originalBitmapDrawable.getIntrinsicHeight() = " + originalBitmapDrawable.getIntrinsicHeight());
       Bitmap scaledBitmap = Bitmap.createScaledBitmap(originalBitmapDrawable.getBitmap(),
```

Comment:
It's pretty useless...

-Jeff

On Thu, Aug 20, 2015 at 11:54 AM, hal <notifications@github.com> wrote:

> In
> appinventor/components/src/com/google/appinventor/components/runtime/util/MediaUtil.java
> <https://github.com/mit-cml/appinventor-sources/pull/522#discussion_r37546516>
> :
>
> >        int scaledWidth = (int) (form.deviceDensity() * originalBitmapDrawable.getIntrinsicWidth());
> >        int scaledHeight = (int) (form.deviceDensity() * originalBitmapDrawable.getIntrinsicHeight());
> > +      Log.d(LOG_TAG, "form.deviceDensity() = " + form.deviceDensity());
> > +      Log.d(LOG_TAG, "originalBitmapDrawable.getIntrinsicWidth() = " + originalBitmapDrawable.getIntrinsicWidth());
> > +      Log.d(LOG_TAG, "originalBitmapDrawable.getIntrinsicHeight() = " + originalBitmapDrawable.getIntrinsicHeight());
> >        Bitmap scaledBitmap = Bitmap.createScaledBitmap(originalBitmapDrawable.getBitmap(),
>
> Do we want to try to catch an error here, or is that useless?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/522/files#r37546516>.
>



-- 
_______________________________________________________________________
Jeffrey I. Schiller
Information Systems and Technology – MIT App Inventor
Massachusetts Institute of Technology
77 Massachusetts Avenue  Room E17-110A, 32-392
Cambridge, MA 02139-4307
617.910.0259 - Voice
jis@mit.edu
http://jis.qyv.name
http://appinventor.mit.edu
_______________________________________________________________________